File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.java
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        Set<Integer> set1 = new HashSet<>();
        Set<Integer> set2 = new HashSet<>();  // Corrected to 'Set<Integer> set2'

        // Adding elements from nums1 to set1
        for (int n : nums1) {
            set1.add(n);
        }

        // Check for common elements in nums2 and add them to set2
        for (int m : nums2) {
            if (set1.contains(m)) {
                set2.add(m);  // Adding the common element to set2
            }
        }

        // Convert set2 to an array
        int[] result = new int[set2.size()];
        int index = 0;
        for (int num : set2) {
            result[index++] = num;
        }

        return result;
    }
}

File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.py
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        
        set1 = set(nums1)  # Convert nums1 to a set for fast lookups
        set2 = set()        # Initialize an empty set to store the intersection

        # Loop through each element in nums2
        for num in nums2:   # Corrected variable name from 'num2' to 'nums2'
            if num in set1:  # Check if the element exists in set1
                set2.add(num)  # If yes, add it to the result set

        # Return the result as a list
        return list(set2)
